pkg/aws/sts/cache.go

```diff
@@ -117,7 +119,16 @@ func (c *credentialsCache) CredentialsForRole(ctx context.Context, identity *Rol
 	cacheMiss.Inc()
 
 	issue := func() (interface{}, error) {
-		credentials, err := c.gateway.Issue(ctx, identity.ARN, c.sessionName, c.sessionDuration)
+		sessionName := c.getSessionName(identity)
+
+		stsIssueRequest := &STSIssueRequest{
+			RoleARN:         identity.Role.ARN,
+			SessionName:     sessionName,
```

Comment:
Not sure why it would use `c.getSessionName` rather than `identity.SessionName`?

Reply:
Ah, just seen it looks like there's a default later. I think it's ok to pull that into the `RoleIdentity` type?

Reply:
As discussed this makes sense to leave as we need the default session name.